docs/proposals/001-k8s-status.md

```diff
@@ -0,0 +1,433 @@
+
+## Rationale
+<!--
+This section COULD be as short or as long as needed. In the appropriate amount
+of detail, you SHOULD explain how this proposal improves k8s-snap, what is the
+problem it is trying to solve and how this makes the user experience better.
+
+You can do this by describing user scenarios, and how this feature helps them.
+You can also provide examples of how this feature may be used.
+-->
+
+The current `k8s status` command returns information about the control plane
+nodes, and then mimics the output of the `k8s get` command. This is not useful,
+as this is by definition the "spec" of what the built-in features should be, not
+their status.
+
+If anything happens and deploying a built-in feature fails, then `k8s status`
+does not present this information in any way to the user. Further, the only way
+to debug this is to look for logs in the `k8sd` service, which users might not
+be familiar with.
+
+Instead, we want the `k8s status` command to return meaningful information about
+the current _deployment status_ of each feature. We want to replace the
+following output (taking the `dns` feature as an example):
+
+```bash
+$ k8s status
+...
+dns:
+  enabled: true                         # these fields comes from the cluster
+  service-ip: 10.152.183.10             # configuration. enabled=true does not
+  cluster-domain: cluster.local         # mean that 'dns' is actually available
+```
+
+With something like this:
+
+```bash
+$ k8s status
+...
+dns:
+  enabled: true/false                   # true if successful, false otherwise
+  message: CoreDNS deployed
+```
+
+## User facing changes
+<!--
+This section MUST describe any user-facing changes that this feature brings, if
+any. If an API change is required, the affected endpoints MUST be mentioned. If
+the output of any k8s command changes, the difference MUST be mentioned, with a
+clear example of "before" and "after".
+-->
+
+### `k8s status` command
+
+The output format of `k8s status` will change.
+
+#### Old output
+
+```bash
+$ k8s status
+status: ready
+high-availability: no
+datastore:
+  type: k8s-dqlite
+  voter-nodes:
+    - 10.87.24.119:6400
+  standby-nodes: none
+  spare-nodes: none
+network:
+  enabled: true
+dns:
+  enabled: true
+  cluster-domain: cluster.local
+  service-ip: 10.152.183.104
+  upstream-nameservers:
+  - /etc/resolv.conf
+ingress:
+  enabled: false
+  default-tls-secret: ""
+  enable-proxy-protocol: false
+load-balancer:
+  enabled: false
+  cidrs: []
+  l2-mode: false
+  l2-interfaces: []
+  bgp-mode: false
+  bgp-local-asn: 0
+  bgp-peer-address: ""
+  bgp-peer-asn: 0
+  bgp-peer-port: 0
+local-storage:
+  enabled: true
+  local-path: /var/snap/k8s/common/rawfile-storage
+  reclaim-policy: Delete
+  default: true
+gateway:
+  enabled: true
+```
+
+#### New output
+
+With `--output-format=plain`, a simple overview is shown, with `yaml` or `json`
+more fields are shown:
+
+```bash
+$ k8s status
+ready: true
```

Comment:
What does the `ready:true` here mean?
It seems repeated by the `status: ready` one line below.

Reply:
ah, let me revert. while we are at it, i thought of changing `status: ready`/`status: not ready` to `ready: true/false`

but let me take this out, it's irrelevant for this proposal